ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/ConnectorBindingsRhino.Selection.cs

```diff
@@ -76,36 +76,30 @@ public override List<ISelectionFilter> GetSelectionFilters()
 
   public override void SelectClientObjects(List<string> objs, bool deselect = false)
   {
-    var isPreview = PreviewConduit != null && PreviewConduit.Enabled ? true : false;
+    var isPreview = PreviewConduit != null && PreviewConduit.Enabled;
 
     foreach (var id in objs)
     {
-      RhinoObject obj = null;
-      try
+      if (Utils.GetGuidFromString(id, out Guid guid))
       {
-        obj = Doc.Objects.FindId(new Guid(id)); // this is a rhinoobj
-      }
-      catch
-      {
-        continue; // this was a named view!
-      }
-
-      if (obj != null)
-      {
-        if (deselect)
+        if (Doc.Objects.FindId(guid) is RhinoObject obj)
         {
-          obj.Select(false, true, false, true, true, true);
+          if (deselect)
+          {
+            obj.Select(false, true, false, true, true, true);
```

Comment:
I would remove the if clause and make one call to obj.Select passing !deselect, for clarity I'd probably set a new variable with a useful name if possible to !deselect and pass that in.

Reply:
But appreciate this is pre-existing

Reply:
Yes preexisting, but a good and easy change to make